server2/Arcadia.Assistant/Arcadia.Assistant.WorkHoursCredit/Notification/WorkHoursCreditNotification.cs

```diff
@@ -0,0 +1,301 @@
+            IWorkHoursCreditApproveNotificationConfiguration workHoursCreditApproveNotificationConfiguration,
+            IWorkHoursCreditCancelNotificationConfiguration workHoursCreditCancelNotificationConfiguration,
+            ILogger<WorkHoursCreditNotification> logger)
+        {
+            this.notifications = notifications;
+            this.dbFactory = dbFactory;
+            this.employeeService = employeeService;
+            this.organizationService = organizationService;
+            this.notificationConfigurations = new Dictionary<string, INotificationConfiguration>
+            {
+                {
+                    WorkHoursCreditNotificationTemplate.WorkHoursCreditCreated,
+                    workHoursCreditCreateNotificationConfiguration
+                },
+                {
+                    WorkHoursCreditNotificationTemplate.WorkHoursCreditApproveRequire,
+                    workHoursCreditApproveRequireNotification
+                },
+                {
+                    WorkHoursCreditNotificationTemplate.WorkHoursCreditApproved,
+                    workHoursCreditApproveNotificationConfiguration
+                },
+                {
+                    WorkHoursCreditNotificationTemplate.WorkHoursCreditCancelled,
+                    workHoursCreditCancelNotificationConfiguration
+                }
+            };
+            this.logger = logger;
+        }
+
+        public async Task SendCreateWorkHoursCreditNotification(
+            EmployeeId employeeId, WorkHoursChangeType changeType, DateTime date, DayPart dayPart,
+            CancellationToken cancellationToken)
+        {
+            this.logger.LogDebug(
+                "Send work hours credits new request {Type} notification (employee id={EmployeeId}.",
+                changeType, employeeId);
+
+            var employee = await this.GetEmployeeMetadataAsync(employeeId, cancellationToken);
+            if (employee == null)
+            {
+                this.logger.LogError("Can't find employee for new request notification.");
+                throw new ArgumentException("Employee missed in database.");
+            }
+
+            try
+            {
+                var context = this.GetNotificationContext(employee!, changeType, date, dayPart);
+                await this.notifications.Send(new[] { employeeId }, this.CreateNotificationMessage(
+                        WorkHoursCreditNotificationTemplate.WorkHoursCreditCreated,
+                        new NotificationDataPresenter(
+                            this.notificationConfigurations[WorkHoursCreditNotificationTemplate.WorkHoursCreditCreated],
+                            context)),
+                    cancellationToken);
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError(e,
+                    "Send Workhours Credit notification error");
+                throw;
```

Comment:
I don't think those should be rethrown. Not here, not in vacations. Ultimately it'll propagate up to API level, which is probably not desirable. If SMTP server returned an error, we don't want the whole operation to be marked as a failure. This is for Vacations too.